run_algorithm

```diff
@@ -0,0 +1,295 @@
+
+    ONE WAY ANOVA
+
+        ./run_algorithm -a one_way_anova -x ppmicategory -y lefthippocampus \
+            -d ppmi1 -m dementia:0.1
+
+    LINEAR REGRESSION
+
+        ./run_algorithm -a linear_regression -y righthippocampus \
+            -x lefthippocampus leftamygdala -d ppmi0 -m dementia:0.1
+
+    LINEAR REGRESSION CROSS-VALIDATION
+
+        ./run_algorithm -a linear_regression_cv -y righthippocampus \
+            -x lefthippocampus leftamygdala -d ppmi0 -m dementia:0.1 -p n_splits 4
+
+    LOGISTIC REGRESSION
+
+        ./run_algorithm -a logistic_regression -y ppmicategory \
+            -x lefthippocampus leftamygdala -d ppmi0 ppmi1 -m dementia:0.1 \
+            -p positive_class PD
+"""
+
+import json
+import pydoc
+import sys
+from argparse import ArgumentParser
+
+import requests
+
+ALGORITHMS_URL = "http://127.0.0.1:5000/algorithms/"
+
+
+def parse_argv(argv):
+    parser = ArgumentParser(
+        prog=f"{sys.argv[0]}",
+        epilog=f"To see full documentation run {sys.argv[0]} --doc.",
+    )
+    required_when_tty = sys.stdin.isatty()
+    parser.add_argument(
+        "-a",
+        "--algorithm",
+        required=True,
+        help="Algorithm name",
+    )
+    parser.add_argument(
+        "-y",
+        required=required_when_tty,
+        nargs="+",
+        help="Y variables, multiple values",
+    )
+    parser.add_argument(
+        "-x",
+        required=False,
+        nargs="+",
+        help="X variables, multiple values",
+    )
+    parser.add_argument(
+        "-d",
+        "--datasets",
+        required=required_when_tty,
+        nargs="+",
+        help="Datasets, multiple values",
+    )
+    parser.add_argument(
+        "-m",
+        "--data-model",
+        required=required_when_tty,
+        help="Data model, full name NAME:VERSION",
+    )
+    parser.add_argument(
+        "-p",
+        "--parameter",
+        required=False,
+        nargs=2,
+        action="append",
+        dest="parameters",
+        help="Parameter as key value pair. Can repeat '-p key value' multiple times.",
+    )
+    parser.add_argument(
+        "-n",
+        "--dry-run",
+        required=False,
+        action="store_true",
+        help="Print request object without making actual request.",
+    )
+    return parser.parse_args(argv)
+
+
+def guess_type_and_cast(value: str):
+    """Casts values passed in the command line by guessing their type.
+
+    Types supported: bool, int, float, str.
+    """
+    try:
+        return int(value)
+    except ValueError:
+        pass
+    try:
+        return float(value)
+    except ValueError:
+        pass
+    if value == "True":
+        return True
+    if value == "False":
+        return False
+    return value
+
+
+def get_commandline_request_data(cl_args):
+    """Formats an algorithm request data object from command line args."""
+    parameters = {
+        key: guess_type_and_cast(value) for key, value in cl_args.parameters or []
```

Comment:
The 'or  []'  is to catch the case of cl_args.parameters being None?

Reply:
Yes, the default value when no parameters are passed is `None`. You can change the default if you want in the `add_argument` method, but passing a mutable default, like `[]` is usually a [bad idea in Python](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments).